File: C++/M01_ProgrammingBasics/L03_ConditionalStatementsAdvanced/Exercises/Solutions/P03_NewHouse.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {

	double rose = 5;
	double dahlia = 3.8;
	double tulip = 2.8;
	double daffodil = 3;
	double gladiolus = 2.5;

	string type, strQuantity, strBudget;
	getline(cin, type);
	getline(cin, strQuantity);
	int quantity = stoi(strQuantity);
	getline(cin, strBudget);
	int budget = stoi(strBudget);

	double price;
	double percentage = 1;
	if (type == "Roses") {
		price = rose;
		if (80 < quantity) {
		    percentage = 0.9;
		}
	} else if (type == "Dahlias") {
		price = dahlia;
		if (90 < quantity) {
			percentage = 0.85;
		}
	} else if (type == "Tulips") {
		price = tulip;
		if (80 < quantity) {
			percentage = 0.85;
		}
	} else if (type == "Narcissus") {
		price = daffodil;
		if (quantity < 120) {
			percentage = 1.15;
		}
	} else if (type == "Gladiolus") {
		price = gladiolus;
		if (quantity < 80) {
			percentage = 1.2;
		}
	}

	double totalPrice = quantity * price * percentage;

	cout.setf(ios::fixed);
	cout.precision(2);

	double diff = abs(totalPrice - budget);
	if (totalPrice <= budget) {
		cout << "Hey, you have a great garden with " << quantity << " " << type << " and " << diff << " leva left.";
	} else {
		cout << "Not enough money, you need " << diff << " leva more.";
	}

	return 0;
}

File: Python/M01_ProgrammingBasics/L03_ConditionalStatementsAdvanced/Exercises/Solutions/P03_NewHouse.py
flowers = input()
qty = int(input())
budget = int(input())

price = 0

Roses = 5
Dahlias = 3.8
Tulips = 2.8
Narcissus = 3
Gladiolus = 2.5

if flowers == "Roses":
    if qty > 80:
        price = Roses * qty * 0.9
    else:
        price = Roses * qty
elif flowers == "Dahlias":
    if qty > 90:
        price = Dahlias * qty * 0.85
    else:
        price = Dahlias * qty
elif flowers == "Tulips":
    if qty > 80:
        price = Tulips * qty * 0.85
    else:
        price = Tulips * qty
elif flowers == "Narcissus":
    if qty < 120:
        price = Narcissus * qty * 1.15
    else:
        price = Narcissus * qty
elif flowers == "Gladiolus":
    if qty < 80:
        price = Gladiolus * qty * 1.2
    else:
        price = Gladiolus * qty

if price > budget:
    print(f'Not enough money, you need {price - budget:.2f} leva more.')
else:
    print(f'Hey, you have a great garden with {qty} {flowers} and {budget - price:.2f} leva left.')